src/library.cpp

```diff
@@ -157,6 +157,90 @@ bool Library::removeBookmark(const std::string& zimId, const std::string& url)
   return false;
 }
 
+int Library::migrateBookmarks() {
+  std::set<std::string> sourceBooks;
+  {
+    std::lock_guard<std::recursive_mutex> lock(m_mutex);
+    for(auto& bookmark:m_bookmarks) {
+      if (m_books.find(bookmark.getBookId()) == m_books.end()) {
+        sourceBooks.insert(bookmark.getBookId());
+      }
+    }
+  }
+  int changed = 0;
+  for(auto& sourceBook:sourceBooks) {
+    changed += migrateBookmarks(sourceBook);
+  }
+  return changed;
+}
+
+std::string Library::getBestTargetBookId(const Bookmark& bookmark) const {
+  // Search for a existing book with the same name
+  auto book_filter = Filter();
+  if (!bookmark.getBookName().empty()) {
+    book_filter.name(bookmark.getBookName());
+  } else {
+    // We don't have a name stored (older bookmarks)
+    // Fallback on title (All bookmarks should have one, but let's be safe against wrongly filled bookmark)
+    if (bookmark.getBookTitle().empty()) {
+        // No bookName nor bookTitle, no way to find target book.
+        return "";
+    }
+    book_filter.query("title:\""+bookmark.getBookTitle() + "\"");
```

Comment:
What if the book title contains a double-quote character? A test case is due.

Reply:
Done